openpype/hosts/blender/plugins/publish/extract_blend.py

```diff
@@ -1,56 +1,265 @@
 import os
+from pathlib import Path
+from typing import List, Set, Tuple
 
 import bpy
 
-from openpype.pipeline import publish
+from openpype.pipeline import (
+    legacy_io,
+    publish,
+)
+from openpype.lib import source_hash
+from openpype.hosts.blender.api import plugin, get_compress_setting
+from openpype.settings.lib import get_project_settings
 
 
 class ExtractBlend(publish.Extractor):
-    """Extract a blend file."""
+    """Extract the scene as blend file."""
 
     label = "Extract Blend"
     hosts = ["blender"]
-    families = ["model", "camera", "rig", "action", "layout"]
+    families = ["model", "camera", "rig", "layout", "setdress"]
     optional = True
 
+    pack_images = True  # TODO must be a OP setting
+
     def process(self, instance):
         # Define extract output file path
-
         stagingdir = self.staging_dir(instance)
         filename = f"{instance.name}.blend"
-        filepath = os.path.join(stagingdir, filename)
+        filepath = Path(stagingdir, filename)
+
+        # If paths management, make paths absolute before saving
+        project_name = instance.data["projectEntity"]["name"]
+        project_settings = get_project_settings(project_name)
```

Comment:
Project settings are already available on context
```suggestion
        project_settings = instance.context.data["project_settings"]
```